docs/src/model_creation/dsl_basics.md

```diff
@@ -264,6 +264,17 @@ Catalyst comes with the following predefined functions:
 - The repressive Hill function: $hillr(X,v,K,n) = v * (K^n)/(X^n + K^n)$.
 - The activating/repressive Hill function: $hillar(X,Y,v,K,n) = v * (X^n)/(X^n + Y^n + K^n)$.
 
+### [Registration of non-algebraic functions](@id dsl_description_nonconstant_rates_function_registration)
+Previously we showed how user-defined functions [can be used in rates directly](@ref dsl_description_nonconstant_rates_available_functions). For functions containing more complicated syntax (e.g. `for` loops or `if` statements), we must add an additional step: registering it using the `@register_symbolic` macro. Below we define a function which output depends on whether `X` is smaller or larger than a threshold value. Next, we register it using `@register_symbolic`, after which we can use it within the DSL.
+```@example dsl_basics
+threshold_func(X) = (X < 10) ? X : 10.0
+@register_symbolic threshold_func(X)
+rn = @reaction_network begin
+    threshold_func(X), 0 --> X
+    d, X --> 0
+end
```

Comment:
currently models in this file are named `rn_1` and so on. This didn't hurt when it was written in the beginning, but annoying to maintain and not sure if there is a point to it. I think I will go in and change them all to `rn`, but not in this PR.

Reply:
`thresfunc(X) = (X < 10) ? X : 10.0` is maybe not the best example, since this is essentially `min(X,10)` which is supported normally. Still, I liked it because it is something one could plausible use in some situations. Could replace it with `thresfunc(X) = (X < 10) ? 1.0 : 2.0`, which is a bit less relevant (but really do requires `@register_symbolic`).

Potentially we should add a note that this kind of discontinuities might affect numeric performance a bit.

Reply:
But it should use `ifelse` instead to allow AD to work more generally?

Reply:
good point. I will think if I can come up with something else so we don't have to get into stuff like this in the most basic tutorial.

Reply:
I have changed it to
```julia
weirdfunc(x) = round(x) + 2.0
@register_symbolic weirdfunc(X)
rn = @reaction_network begin
    weirdfunc(X), 0 --> X
    d, X --> 0
end
```